### ISSUE TITLE
OpenStack - additional storage should be mounted as additional volume

### ISSUE THREAD

**Issue by keiranmraine** [VertebrateResequencing/wr#66]
Working through some issues with ISG.

@jbeal-work indicated that having the OS build on the attached CEPH volume and not booting from the local SSD is a bad idea as it increases load.

> I do boot a new machine and attach a volume so I have a second device as /data01

**Comment by sb10** [comment]
I'll be looking at volume possibilities to get the most performant implementation of #54, but for now I'll note that if I change this for everything I'll probably lose auto-deletion of these volumes on server destruction. I'll try and do it "manually", of course, but there's always that possibility that my separate volume deletion request doesn't go through.

**Comment by jbeal-work** [comment]
When you build the volume with the machine creation you can  flag the volume as auto delete. I can find some ruby code.

**Comment by sb10** [comment]
Yes, that's what I'm doing. But it is that method that is resulting in the OS being on the CEPH volume. Can you somehow change the OpenStack configuration so that it does what you want in response to an auto-delete request?

**Comment by keiranmraine** [comment]
@sb10, the ability for volumes to not delete on failure is desirable as an option too

**Comment by jbeal-work** [comment]
This is the ruby I have which may guide in the right direction.

    # If only it was simple
    # http://docs.openstack.org/developer/nova/block_device_mapping.html
    # Who thought it was a sane idea to have size being in Gigabytes.
    if ws_values.has_key?(:extra_disk_sizes) && ws_values[:extra_disk_sizes] != "0"
      log(:info, "#{@log_prefix}: extra_disk_sizes #{ws_values[:extra_disk_sizes].inspect},#{ws_values[:extra_disk_sizes].class}")
      log(:info, "#{@log_prefix}: current_options -#{prov.get_option(:clone_options).inspect}'")
      block_device_mapping_v2 = []
      boot_disk = find_object_for("template_cloud",template.name).uid_ems
      log(:info, "#{@log_prefix}:find the disk for boot -#{boot_disk}-")
      disk_config = {
        :boot_index => 0,
        :uuid => boot_disk,
        :source_type => "image",
        # For the boot disk a size needs to be specified but it is over ridden by the flavor 
        :volume_size => 20, 
        :destination_type => "local",
        :delete_on_termination => true }
      block_device_mapping_v2 << disk_config
      ws_values[:extra_disk_sizes].gsub(/[\[\]\"]/,"").split(",").each do |disk_size|
        log(:info, "#{@log_prefix}: The extradisk_size is <#{disk_size}>")
        disk_config = {
          :source_type => "blank",
          :volume_size => disk_size.to_i,
          :name => "jb23 testname",
          :destination_type => "volume",
          :delete_on_termination => false  }
        block_device_mapping_v2 << disk_config
      end
      prov.set_option( :clone_options, {
                          :image_ref => boot_disk,
                          :block_device_mapping_v2 => block_device_mapping_v2 } )
      log(:info, "#{@log_prefix}: The block_device_mapping_v2 is <#{block_device_mapping_v2.inspect}>")
    end

**Comment by sb10** [comment]
What is the end result of that ruby? What do you get in OpenStack?

**Comment by jbeal-work** [comment]
I get a machine with the os on local disk and a second disk provisioned via shared storage ( ceph )
